lib/jekyll/collection.rb

```diff
@@ -58,7 +58,7 @@ def read
       filtered_entries.each do |file_path|
         full_path = collection_dir(file_path)
         next if File.directory?(full_path)
-        if Utils.has_yaml_header? full_path
+        if Utils.has_yaml_header?(full_path) || read_yaml_file?(full_path)
```

Comment:
This is something I had a concern about. I wanted to ensure taht yml files being processed was not just for yaml files with `---` at the top, but what I've seen a lot of yaml files be, which is just the data. 

I am thinking though that perhaps since we want the data to be read to the `site.data` that we might want to eliminate where I check below if its meant to be written, and instead just check that its a yml file.